katrain/core/sgf_parser.py

```diff
@@ -385,7 +386,10 @@ def parse_file(cls, filename, encoding=None) -> SGFNode:
                     if match:
                         encoding = match[1].decode("ascii", errors="ignore")
                     else:
-                        encoding = cls.DEFAULT_ENCODING
+                        encoding = chardet.detect(bin_contents[:300])['encoding']
```

Comment:
why 300? is it too slow otherwise?

Reply:
https://pypi.org/project/cchardet/
cchardet seems to offer a fast character detection function .1000x faster than chardet.

Reply:
@xiaoyifang I'm a bit hesitant to include more things that need compiling as they can give problems with os compatibility.

Reply:
@sanderland this first 300 bytes choice is copied from Sabaki sgf parser. My simple timeit test shows that detecting a typical 2KB SGF file on my PC is ~12 ms while detecting only first 300 bytes is ~ 3ms. The performance diff is negligible for typical SGF file. But based on my test, since the "encoded" words mainly appear in the beginning of SGF file, chardet shows exact same "confidence" detecting full or only first 300 bytes.

So in summary I think that optimization should be OK.